### PR TITLE
Improve Integer Rounding Accuracy

### PR DESCRIPTION
**Pull request by Geotale**
Changes integer rounding to more closely meet the documentation (rounding before doing any bounds checks). All this really does is make sure some exception bits won't be erroneously set. This depends on the rounding mode, so with rounding to even, (0x7fffffff, 0x7fffffff.8) now works properly without setting these bits!!
This change also uses some standard functions for rounding instead of what was previously used (due to previously using them to cast to an s32 directly, while now needing that double value just a little bit longer)
Finally, due to comparing with a rounded double rather than a non-rounded one, a single >0x7fffffff was changed to >=0x80000000 because it ~~looks nicer~~ gives better codegen on Aarch64 ^^

### CODE REVIEW

**Comment by Geotale** [comment]
RoundEven not found...? (And lint issue)

**Comment by BhaaLseN** [comment]
Yep, we're usually held back by a few platforms lagging behind in updating their toolset, so the most common denominator is C++20 at this point. Feel free to keep this open (and rebase it every now and then) so we can benefit from it once C++23 rolls around on every platform.

Please do address the lint notes though, you can run the `Tools/lint.sh` script to do it for you.

**Review by JosJuice — COMMENTED** [review]
Code changes look good overall, but please squash the commits into one.

Also, the commit message (of the first commit) contains a lot less detail than the PR description. In particular, what the "now able to change a single value" part refers to is unclear. It would be better to replace this commit message with the message you've already written for the PR description.

File: Source/Core/Core/PowerPC/Interpreter/Interpreter_FloatingPoint.cpp [hunk 1]
```diff
@@ -32,16 +33,56 @@ void SetFI(PowerPC::PowerPCState& ppc_state, u32 FI)
...
   bool exception_occurred = false;
 
+  // To reduce complexity, this takes in a rounding mode in a switch case,
+  // rather than always judging based on the emulated CPU rounding mode
+  switch (rounding_mode)
+  {
+  case RoundingMode::Nearest:
+    // On generic platforms, the rounding should be assumed to be ties to even
+    // For targeted platforms this would work for any rounding mode,
+    // but it's mainly just kept in to replace roundeven
```
[the remark this answers is not in the document]
Comment (Dentomologist) [reply Source/Core/Core/PowerPC/Interpreter/Interpreter_FloatingPoint.cpp:70]: I've been wondering, is roundeven even going to be in c++23? It's defined in c23, but according to the accepted answer [here](https://stackoverflow.com/questions/77418146/which-version-of-the-c-standard-library-does-the-c23-standard-incorporate) c++23 will only include the c17 standard library.

**Comment by Geotale** [comment]
I didn't use the exact changes, but I'm hoping these suffice for everyone's needs ^^ (I'll definitely update the commit message esp in the squash as well soon!!)

**Comment by JosJuice** [comment]
Yes, I'm happy with these comments.

**Comment by Geotale** [comment]
In that case commits squashed :D (with commit message also reflecting every change I hope ^^)

**Comment by JosJuice** [comment]
Just one more thing. Could you word wrap the commit message so no line is longer than 72 characters, as per git conventions?

**Comment by JMC47** [comment]
Once the final comment is addressed, this can be merged.

**Comment by Geotale** [comment]
Changed commit message, I had to reorganize it a bit but I hope it follows guidelines better now ^^

**Pull request merged**